### ISSUE TITLE
[question] Why is the columns shifting and no longer aligned after exporting as Excel?

### ISSUE THREAD

**Issue by aiampogi** [mui/mui-x#13294]
### The problem in depth

Using MUI-X Premium Datagrid v7.5.1, after exporting as Excel, the columns become unaligned and shifts whenever you scroll horizontally.

It can be reproduced using the example in the [docs](https://mui.com/x/react-data-grid/export/#using-a-web-worker)

Here's my recording.
![excelexportingshiftscolumns](https://github.com/mui/mui-x/assets/2859127/869f1cd1-a077-4ecd-84f5-bf152276023c)

### Your environment

<details>
  <summary>`npx @mui/envinfo`</summary>

```
  Browser: Microsoft Edge Version 125.0.2535.67 (Official build) (64-bit)
  Output from `npx @mui/envinfo` goes here:
  
    System:
    OS: Windows 11 10.0.22631
  Binaries:
    Node: 20.9.0 - C:\Program Files\nodejs\node.EXE
    npm: 10.1.0 - C:\Program Files\nodejs\npm.CMD
    pnpm: 8.15.5 - ~\AppData\Local\pnpm\pnpm.EXE
  Browsers:
    Chrome: Not Found
    Edge: Chromium (123.0.2420.97)
  npmPackages:
    @emotion/react: ^11.11.1 => 11.11.1
    @emotion/styled: ^11.11.0 => 11.11.0
    @mui/base: ^5.0.0-beta.32 => 5.0.0-beta.40
    @mui/core-downloads-tracker:  5.15.16
    @mui/icons-material: ^5.11.11 => 5.13.7
    @mui/lab: ^5.0.0-alpha.128 => 5.0.0-alpha.135
    @mui/material: ^5.15.16 => 5.15.16
    @mui/private-theming:  5.15.14
    @mui/styled-engine:  5.15.14
    @mui/system:  5.15.15
    @mui/types:  7.2.14
    @mui/utils:  5.15.14
    @mui/x-data-grid:  7.5.1
    @mui/x-data-grid-premium: ^7.5.1 => 7.5.1
    @mui/x-data-grid-pro:  7.5.1
    @mui/x-license:  7.2.1
    @types/react:  18.2.14
    react: ^18.2.0 => 18.2.0
    react-dom: ^18.2.0 => 18.2.0
    typescript: ^4.9.4 => 4.9.5
```
</details>

**Search keywords**: export as excel shifting columns

**Comment by romgrk** [comment]
Duplicate of #12641? The fix will be released this week.

**Comment by aiampogi** [comment]
cool thanks. I'll close this.

**Comment by aiampogi** [comment]
what version will it be fixed?

**Comment by romgrk** [comment]
The upcoming one in the v7 branch. Either 7.6.0 or 7.5.2 depending on what changes are included.

**Comment by aiampogi** [comment]
thanks.

**Comment by aiampogi** [comment]
as per @romgrk it'll be fixed either in 7.6.0 or 7.5.2
